core/src/test/java/org/opensearch/sql/analysis/AnalyzerTestBase.java

```diff
@@ -158,19 +176,24 @@ protected LogicalPlan analyze(UnresolvedPlan unresolvedPlan) {
 
   private class DefaultDataSourceService implements DataSourceService {
 
-    private StorageEngine storageEngine = storageEngine();
-    private final DataSource dataSource
-        = new DataSource("prometheus", DataSourceType.PROMETHEUS, storageEngine);
+    private final DataSource opensearchDataSource = new DataSource(DEFAULT_DATASOURCE_NAME,
+        DataSourceType.OPENSEARCH, storageEngine());
+    private final DataSource prometheusDataSource
```

Comment:
Maybe: add a tests with another datasource too?

Reply:
Any test case should be covered in analyer? Analyzer should not differenate different datasource.